src/main/java/com/example/springlabs/repositories/CategoryRepository.java

```diff
@@ -1,21 +1,30 @@
 package com.example.springlabs.repositories;
 
 import com.example.springlabs.model.Category;
-import java.util.Collection;
+import jakarta.transaction.Transactional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
 import java.util.List;
 import java.util.Optional;
 
-public interface CategoryRepository {
-    List<Category> getCategories();
-
-    void addCategory(Category category);
+public interface CategoryRepository  extends JpaRepository<Category, Long> {
 
-    Category getCategoryById(long id);
+    //find by id - @NamedEntity
+    @Query(nativeQuery = true, name = "Categories.getCategoryById")
+    Category getCategoryById(@Param("id") Long id);
 
-    Optional<Category> updateCategory(long id, Category newCategory,
-        Collection<Category> subcategories);
+    // read all - jpa
+    List<Category> findAll();
 
-    void deleteCategory(long id, Collection<Category> subcategories);
+    // Delete - @Query
+    @Modifying
+    @Transactional
+    @Query("DELETE FROM categories u WHERE u.id = :id")
+    void deleteCategory(@Param("id") Long id);
 
-    Optional<Category> getCategoryByName(String name);
+    @Query(value = "SELECT c FROM categories c WHERE c.name = :name", nativeQuery = true)
```

Comment:
For me, this does not work. I changed `SELECT c FROM` to `SELECT * FROM` and it worked